README.md

```diff
@@ -1,87 +1,86 @@
-# biconomy-paymasters
+# Biconomy Paymasters 🌐
 
-### Playground for ERC4337 Paymasters built with :heart_eyes: by Biconomy 
-#### Some examples 
+Welcome to the `biconomy-paymasters` repository! Paymaster contracts enable seamless transaction fee handling in decentralized applications. 🛠️
 
-1. Singleton Verifying Paymaster: Acts as a sponsorship paymaster and lets Dapps manage deposit without deploying a new one for each Dapp.
+## What is Biconomy Paymasters? 🤔
 
-2. Token Paymaster: Extended version of Verifying Paymaster which can accept fees from user by withdrawing ERC20 tokens
+Biconomy Paymasters are smart contracts that abstract away the complexity of gas fees for end-users. By utilizing these contracts, developers can offer their users gasless transactions or the ability to pay for gas in ERC20 tokens. 🚀
 
-#### Other WIP
+- **Sponsorship Paymaster**: Allows transactions without end-users needing to pay for gas, enhancing UX.
+- **Token Paymaster**: Provides the ability to pay for transactions with ERC20 tokens.
 
-Fiat Paymaster
+## Features 🌟
 
-Deposit Paymaster
+- Simplified transaction fee handling.
+- ERC 4337 Account Abstraction compliant.
+- Multi-token support for gas payments.
 
-Custom versions of above Verifying (allow Dapp deposit sponsorship in different tokens) and Token Paymaster (acts as deposit paymaster also)
+### ERC20 Token Paymaster
 
+![ERC20 Token Paymaster](./assets/readme/erc20-token-gas-payment-flow.png)
 
+- ERC20 Token Paymaster helps users pay for their transactions using ERC20 tokens.
+- Users initiate a transaction using an ERC20 token.
+- Paymaster validates the transaction and forwards it to the network while handling necessary fee conversions.
```

Comment:
Technically Paymaster doesn't forward the transaction. It just pays for the userOp.